packages/@okta/vuepress-site/guides/sign-into-spa/section03/angular/attachcomponentsetc.md

```diff
@@ -1,23 +1,67 @@
-## Attach Components to Routes
-You need to provide these routes in your sample application so that we can sign the user in and handle the callback from Okta. We show you how to set these up below using [Angular Router](https://angular.io/guide/router):
+## Provide the Login and Logout Buttons
 
-- `/`: A default home page to handle basic control of the app.
-- `/implicit/callback`: Handle the response from Okta and store the returned tokens.
+In the relevant location in your application, you want to provide `Login` and `Logout` buttons for the user. The `OktaAuthService` is injected in your component's constructor. You can use the `oktaAuth.isAuthenticated()` method to show/hide the correct button. For example:
 
-## Provide the Login and Logout Buttons
+```javascript
+import { Component } from '@angular/core';
+import { OktaAuthService } from '@okta/okta-angular';
+
+@Component({
+  selector: 'app-root',
+  template: `
+    <button *ngIf="!isAuthenticated" (click)="login()"> Login </button>
+    <button *ngIf="isAuthenticated" (click)="logout()"> Logout </button>
+    <router-outlet></router-outlet>
+  `,
+})
+export class AppComponent {
+  isAuthenticated: boolean;
 
-In the relevant location in your application, you want to provide `Login` and `Logout` buttons for the user. You can show/hide the correct button by using the `oktaAuth.isAuthenticated()` method. For example:
+  constructor(public oktaAuth: OktaAuthService) {
+    // Subscribe to authentication state changes
+    this.oktaAuth.$authenticationState.subscribe(
+      (isAuthenticated: boolean)  => this.isAuthenticated = isAuthenticated
+    );
+  }
 
-{code}
+  async ngOnInit() {
+    // Get the authentication state for immediate use
+    this.isAuthenticated = await this.oktaAuth.isAuthenticated();
+  }
+
+  login() {
+    this.oktaAuth.loginRedirect('/profile');
+  }
+
+  logout() {
+    this.oktaAuth.logout('/');
+  }
+}
+```
 
 ## Create the Callback Handler
 
-In order to handle the redirect back from Okta, you need to capture the token values from the URL. Use `/implicit/callback` as the callback URL and specify the default `OktaCallbackComponent` and declare it in your `NgModule`.
+In order to handle the redirect back from Okta, you need to capture the token values from the callback URL `/implicit/callback` and pass them to the handleAuthentication() method of the OktaAuthService. We have provided `OktaCallbackComponent` which implements this logic.  We show you how to set this up below using [Angular Router](https://angular.io/guide/router):
```

Comment:
In order to handle the redirect back from Okta, you need to capture the token values from the callback URL `/implicit/callback` and pass them to the `handleAuthentication()` method of the `OktaAuthService`. We have provided `OktaCallbackComponent` that implements this logic.  We show you how to set this up below using [Angular Router](https://angular.io/guide/router):